src/server/middleware/authentication.ts

```diff
@@ -0,0 +1,108 @@
+import passport from 'passport'
+import passportLocal, { VerifyFunctionWithRequest } from 'passport-local'
+import bcrypt from 'bcryptjs'
+import { Express } from 'express'
+import { FieldValidators, User, UserStatus, Validator } from '@openforis/arena-core'
+
+import { ExpressInitializer } from '../expressInitializer'
+import { UserServiceServer } from '../../service'
+import { UserResetPasswordRepository, AuthGroupRepository } from '../../repository'
+
+const comparePassword = bcrypt.compareSync
+
+// TODO: Handle these separately
+const ValidationMessages = {
+  emailInvalid: 'validationErrors.user.emailInvalid',
+  userNotFound: 'validationErrors.user.userNotFound',
+  passwordChangeRequired: 'validationErrors.user.passwordChangeRequired',
+}
+
+const _initializeUser = async (_user: User): Promise<User> => {
+  const user = { ..._user }
+  if (user.password) delete user.password
+
+  // Assoc auth groups
+  let userUpdated = {
+    ...user,
+    authGroups: await AuthGroupRepository.fetchUserGroups({ userUuid: user.uuid }),
+  }
+  if (user.status === UserStatus.INVITED) {
+    const expired = !(await UserResetPasswordRepository.existsResetPasswordValidByUserUuid({ userUuid: user.uuid }))
+    userUpdated = {
+      ...userUpdated,
+      invitation: {
+        expired,
+      },
+    }
+  }
+
+  // @ts-ignore fixed by implementing repository method
```

Comment:
why ts-ignore ?

Reply:
```
Type '{ authGroups: void; email: string; hasProfilePicture: boolean; name: string; prefs?: UserPrefs | undefined; status: UserStatus; password?: string | undefined; invitation?: UserInvitation | undefined; uuid: string; props: UserProps; validation?: Validation | undefined; }' is not assignable to type 'User'.
  Types of property 'authGroups' are incompatible.
    Type 'void' is not assignable to type 'AuthGroup[] | undefined'.ts(2322)
    ```

Reply:
and there's no way to fix this?

Reply:
The ts-ignore comment has `fixed by implementing repository method`